src/app/update-user/update-user.component.html

```diff
@@ -8,54 +8,57 @@
           </div>
 
           <div class="form-group">
             <label for="surname">Last name:</label>
-            <input type="text" id="surname" [(ngModel)]="userToEdit.lastName" name="surname" required #surnameModel="ngModel" />
+            <input type="text" id="surname" [(ngModel)]="userToEdit.lastName" name="surname" required #surnameModel="ngModel"  required pattern="[a-zA-Z]+" [ngClass]="{ 'invalid-input': surnameModel.invalid && (surnameModel.dirty || surnameModel.touched) }"/>
             <div *ngIf="surnameModel.invalid && (surnameModel.dirty || surnameModel.touched)" class="error-message">
-              <div *ngIf="surnameModel.hasError('required')">Surname is required.</div>
+              <div *ngIf="surnameModel.hasError('required')" style="color: red;">Surname is required.</div>
+              <div *ngIf="nameModel.hasError('pattern')" style="color: red;">Last name must contain only letters.</div>
             </div>
           </div>
 
         <div class="form-group">
             <label for="jmbg">JMBG:</label>
-            <input type="text" id="jmbg" [(ngModel)]="userToEdit.jmbg" name="jmbg" required pattern="^\d{13}$" #jmbgModel="ngModel" />
+            <input type="text" id="jmbg" [(ngModel)]="userToEdit.jmbg" name="jmbg" required pattern="^\d{13}$" #jmbgModel="ngModel" [ngClass]="{ 'invalid-input': jmbgModel.invalid && (jmbgModel.dirty || jmbgModel.touched) }"  />
             <div *ngIf="jmbgModel.invalid && (jmbgModel.dirty || jmbgModel.touched)" class="error-message">
-              <div *ngIf="jmbgModel.hasError('required')">JMBG is required.</div>
-              <div *ngIf="jmbgModel.hasError('pattern')">Invalid JMBG format (must be 13 digits).</div>
+              <div *ngIf="jmbgModel.hasError('required')" style="color: red;">JMBG is required.</div>
+              <div *ngIf="jmbgModel.hasError('pattern')" style="color: red;">Invalid JMBG format (must be 13 digits).</div>
             </div>
           </div>
 
           <div class="form-group">
             <label for="password">Password:</label>
-            <input type="password" id="password" [(ngModel)]="password" name="password" required #passwordModel="ngModel" />
+            <input type="password" id="password" [(ngModel)]="password" name="password" required #passwordModel="ngModel" required pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$" [ngClass]="{ 'invalid-input': passwordModel.invalid && (passwordModel.dirty || passwordModel.touched) }" />
             <div *ngIf="passwordModel.invalid && (passwordModel.dirty || passwordModel.touched)" class="error-message">
-              <div *ngIf="passwordModel.hasError('required')">Password is required.</div>
+              <div *ngIf="jmbgModel.hasError('required')" style="color: red;">Password is required.</div>
+              <div *ngIf="jmbgModel.hasError('pattern')" style="color: red;">Invalid password format (must be 8 chars long, UC,LC).</div>
             </div>
           </div>
```

Comment:
Ovde nesto nije dobro, crveni mi se sta god da unesem u to polje, a posalje se zahtev